Framework/PythonInterface/mantid/simpleapi.py

```diff
@@ -806,68 +850,72 @@ def ignore_property(name, ignore_regex):
         if _is_workspace_property(prop):
             value_str = prop.valueAsStr
             try:
-                retvals[name]=_api.AnalysisDataService[value_str]
+                retvals[name] = _api.AnalysisDataService[value_str]
             except KeyError:
                 if not prop.isOptional():
-                    raise RuntimeError("Internal error. Output workspace property '%s' on algorithm '%s' has not been stored correctly."
+                    raise RuntimeError("Internal error. Output workspace property '%s' on "
+                                       "algorithm '%s' has not been stored correctly."
                                        "Please contact development team." % (name,  algm_obj.name()))
         else:
             if hasattr(prop, 'value'):
-                retvals[name]=prop.value
+                retvals[name] = prop.value
             else:
-                raise RuntimeError('Internal error. Unknown property type encountered. "%s" on algorithm "%s" is not understood by '
-                       'Python. Please contact development team' % (name, algm_obj.name()))
-
-
+                raise RuntimeError('Internal error. Unknown property type encountered. "%s" '
+                                   'on algorithm "%s" is not understood by '
+                                   'Python. Please contact development team' % (name, algm_obj.name()))
     nvals = len(retvals)
     nlhs = lhs[0]
-    if nlhs > 1 and nvals != nlhs:
+
+    # If we have more than one value but not the same number of values throw
+    if 1 < nlhs != nvals:
```

Comment:
Chaining expressions is something Python encourages you to do so my IDE warns me on this line. I'm not sure whether it is a flake warning but it tends to be flake warnings are a subset of the warnings I get locally. 

How about tidying it up to both remove the warning and make the intention a bit clearer:
Original:
```python
    if nlhs > 1 and nvals != nlhs:
        # There is a discrepancy in the number are unpacking variables
        # Let's not have the more cryptic unpacking error raised
        raise RuntimeError("%s is trying to return %d output(s) but you have provided %d variable(s). "
                           "These numbers must match." % (func_name, nvals, nlhs))
    if nvals > 0:
        ret_type=namedtuple(func_name+"_returns",retvals.keys())
        ret_value=ret_type(**retvals)
        if nvals==1:
            return ret_value[0]
        else:
            return ret_value
    else:
        return None
```
Into something like:
```python
if nlhs == 0:
    return None

if nvals != nlhs:
   if nlhs == 1:
        pass   # This is fine as we do not need to unpack
   else:
               # There is a discrepancy in the number are unpacking variables
        # Let's not have the more cryptic unpacking error raised
        raise RuntimeError("%s is trying to return %d output(s) but you have provided %d variable(s). "
                           "These numbers must match." % (func_name, nvals, nlhs))

# Unpack and return
ret_type=namedtuple(func_name+"_returns",retvals.keys())
ret_value=ret_type(**retvals)
return ret_value[0] if nvals == 1 else ret_value